packages/core/src/types.tsx

```diff
@@ -442,21 +453,26 @@ export type ScreenListeners<
 
 export type RouteConfigComponent<
   ParamList extends ParamListBase,
-  RouteName extends keyof ParamList
+  RouteName extends keyof ParamList,
+  ComponentProps = { route: RouteProp<ParamList, RouteName>; navigation: any },
+  Component =
+    | React.ComponentType<ComponentProps>
+    | React.ComponentType<Partial<ComponentProps>>
+    | React.ComponentType<{}>
 > =
   | {
       /**
        * React component to render for this screen.
        */
-      component: React.ComponentType<any>;
+      component: Component;
       getComponent?: never;
       children?: never;
     }
   | {
       /**
        * Lazily get a React component to render for this screen.
        */
-      getComponent: () => React.ComponentType<any>;
+      getComponent: () => Component;
       component?: never;
       children?: never;
     }
```

Comment:
I'd create a helper type for this to avoid adding these generics since `RouteConfigComponent` is part of the public API

```js
type ScreenComponentType<
  ParamList extends ParamListBase,
  RouteName extends keyof ParamList,
  Props = { route: RouteProp<ParamList, RouteName>; navigation: any }
> =
  | React.ComponentType<Props>
  | React.ComponentType<Partial<Props>>
  | React.ComponentType<{}>;

export type RouteConfigComponent<
  ParamList extends ParamListBase,
  RouteName extends keyof ParamList
> =
  | {
      /**
       * React component to render for this screen.
       */
      component: ScreenComponentType<ParamList, RouteName>;
      getComponent?: never;
      children?: never;
    }
  | {
      /**
       * Lazily get a React component to render for this screen.
       */
      getComponent: () => ScreenComponentType<ParamList, RouteName>;
      component?: never;
      children?: never;
    }
```